test/AutoPodArgs/force_pushconstant.ll

```diff
@@ -0,0 +1,14 @@
+; RUN: clspv-opt -AutoPodArgs -pod-pushconstant %s -o %t.ll
+; RUN: FileCheck %s < %t.ll
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; CHECK: define spir_kernel void @foo(i32 addrspace(1)* %out, i32 %pod) !clspv.pod_args_impl [[MD:![0-9]+]]
+; CHECK: [[MD]] = !{i32 2}
+define spir_kernel void @foo(i32 addrspace(1)* %out, i32 %pod) {
```

Comment:
The `auto-pod-args` pass will make this a POD arg a push constant anyway, so this test doesn't really test the `-pod-pushconstant` override. Maybe needs a `-max-pushconstant-size=1` like the `max_size_prevents_push constants` test so that the pass would pick UBO instead? Though I don't know whether forcing push constants that would exceed the size should actually be an error case...

Reply:
The test you describe would be rejected by the frontend so I'd prefer not to add it. The option requests push constants and the compiler delivers. Yes maybe it would anyway, but I'm not actually too worried about the case.